drakrun/drakrun/draksetup.py

```diff
@@ -475,6 +482,208 @@ def mount(iso_path, domain_name):
+    mc.fget_object(bucket, f"{name}/install.json",
+                   os.path.join(ETC_DIR, InstallInfo._INSTALL_FILENAME))
+
+    logging.info("Downloading VM config")
+    mc.fget_object(bucket, f"{name}/cfg.template", os.path.join(ETC_DIR, "scripts", "cfg.template"))
+
+    # Now we have imported InstallInfo object
+    install_info = InstallInfo.load()
+
+    # Patch ZFS pool name
+    if zpool is not None:
+        install_info.zfs_tank_name = zpool
+        install_info.save()
+
+    storage = get_storage_backend(install_info)
+
+    with tempfile.NamedTemporaryFile() as disk_image:
+        logging.info("Downloading VM disk image")
+        mc.fget_object(bucket, f"{name}/disk.img", disk_image.name)
+
+        logging.info("Importing VM disk")
+        storage.import_vm0(disk_image.name)
+
+    # This could probably use some refactoring
+    # We're duplicating quite a lot of code from install function
+
+    generate_vm_conf(install_info, 0)
+    backend = get_storage_backend(install_info)
+    backend.rollback_vm_storage(0)
+
+    net_enable = int(conf['drakrun'].get('net_enable', '0'))
+    out_interface = conf['drakrun'].get('out_interface', '')
+    dns_server = conf['drakrun'].get('dns_server', '')
+    setup_vm_network(
+        vm_id=0,
+        net_enable=net_enable,
+        out_interface=out_interface,
+        dns_server=dns_server
+    )
+
+    if net_enable:
+        start_dnsmasq(vm_id=0, dns_server=dns_server, background=True)
+
+    cfg_path = os.path.join(VM_CONFIG_DIR, "vm-0.cfg")
+
+    try:
+        subprocess.run(['xl' 'create', cfg_path], check=True)
+    except subprocess.CalledProcessError:
+        logging.exception("Failed to launch VM vm-0")
+        return
+
+    logging.info("Minimal snapshots require postinstall to work correctly")
+    logging.info("Please VNC to the port 5900 to ensure the OS booted correctly")
+    logging.info("After that, execute this command to finish the setup")
+    logging.info("# draksetup postinstall")
+
+
+def do_export_full(mc, bucket, name):
+    """ Perform full snapshot export, symmetric to do_import_full """
+    do_export_minimal(mc, bucket, name)
+
+    with tempfile.NamedTemporaryFile() as compressed_snapshot:
+        # Compress snapshot
+        logging.info("Compressing snapshot.sav")
+        subprocess.check_call(
+            ["gzip", "-c", os.path.join(VOLUME_DIR, "snapshot.sav")],
+            stdout=compressed_snapshot,
+        )
+
+        logging.info("Uploading snapshot.sav.gz")
+        mc.fput_object(bucket, f"{name}/snapshot.sav.gz", compressed_snapshot.name)
+
+    # Upload profiles
+    for file in os.listdir(PROFILE_DIR):
+        logging.info("Uploading profile %s", file)
+        mc.fput_object(bucket, f"{name}/profiles/{file}", os.path.join(PROFILE_DIR, file))
+
+
+def do_import_full(mc, name, bucket, zpool):
+    """ Perform full snapshot import, symmetric to do_export_full """
+    do_import_minimal(mc, name, bucket, zpool)
```

Comment:
This will print message from end of `do_import_minimal`.

Reply:
:+1: